index.js

```diff
@@ -132,16 +132,12 @@ function init (ssb, config) {
           ...m.value.content.recps.filter(isFeed)
         ])
 
-        const record = keystore.group.get(groupId)
-        // if we haven't been in the group since before, register the group
-        if (record == null) {
-          return keystore.group.add(groupId, { key: groupKey, root }, (err) => {
-            if (err) return cb(err)
-            processAuthors(groupId, authors, m.value.author, cb)
-          })
-        } else {
+        // TODO: only add if we're one of the people being added by this msg
```

Comment:
todo

Reply:
adding an `if (recps.includes(ssb.id)){`. Will that work even on restarts? I guess hopefully since the db should re-stream in the same order?

Reply:
although hmm the 2 message types have 2 different streams, so there might be race conditions

Reply:
let's add a test where we restart and see if we're still re-added

Reply:
testing that now and it's passing (at least locally) but i think the risk of a race condition is still too high, should refactor things a bit

Reply:
refactored and passing when i run the file individually, but npm t is failing for some reason

Reply:
hmm this fail seems to have been introduced already by https://github.com/ssbc/ssb-tribes/pull/94/commits/3300be89602ec4452a4868872dfa9cd111c2c12f

looks like
```
tribes.create (opts.addPOBox)
    db.rebuild (ﾉ´ヮ´)ﾉ*:･ﾟ✧

/home/runner/work/ssb-tribes/ssb-tribes/node_modules/levelup/lib/levelup.js:119
      return callback(new OpenError(err))
                      ^
Error [OpenError]: IO error: lock /tmp/ssb-tribes/newPerson/flume/keys/LOCK: already held by process
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/levelup/lib/levelup.js:119:23
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/deferred-leveldown/deferred-leveldown.js:31:21
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
Emitted 'error' event on LevelUP instance at:
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/levelup/lib/levelup.js:60:19
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/levelup/lib/levelup.js:119:14
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14
    [... lines matching original stack trace ...]
    at /home/runner/work/ssb-tribes/ssb-tribes/node_modules/abstract-leveldown/abstract-leveldown.js:38:14 {
  [cause]: undefined
}
    ✔ data.poBoxId

  tribes.excludeMembers
    db.rebuild (ﾉ´ヮ´)ﾉ*:･ﾟ✧

  test count(17) != plan(null)

  Failed tests: There was 1 failure

    ✖ undefined) undefined

  total:     17
  passing:   17
  failing:   1
  668 ms
```

Reply:
hmm seems introduced by the `name` args on the server being used

Reply:
specifically seems to be the newPerson name causing it. let's maybe see if there's a newPerson in an earlier test that doesn't close properly (adding timeouts before starting these servers didn't seem to start)

Reply:
there was a testbot called newPerson in `test/api/add-new-author-listener.test.js`. The test closed pretty quick because of t.plan(6) but the bots closed down very slowly because of our 10sec close delay, which made the bot collide with the one in this test